### PR TITLE
URI Validation Added

### PR DESCRIPTION
**Pull request by noranda**
# URI Validation Added

* Added URI Validation -- checks validity of URI in either single or multi-server input fields (using [this guide](http://codegolf.stackexchange.com/questions/464/shortest-url-regex-match-in-javascript))
* Added `ember-cli-notifications` addon to handle error messages
* Updated dependencies including `ember` and `ember-cli` to conform with `ember-cli-notifications` addon
* Refactored code to use the new HTMLBars syntax instead of `bind-attr`
  * `{{bind-attr class=":fa faIcon"}}` becomes `class="fa {{faIcon}}"`
* Refactored `{{input}}` components using `action="..."` to use `enter="..."`

##### Video
[Bad-URI-Notification-Video](http://youtu.be/BZGxbScpvdw)

##### TODO:
* [ ] Expand validity check to include backend-supported check if URI is a valid FHIR server
* [ ] Continue checking application for bugs related to dependency updates

### CODE REVIEW

**Comment by andrequina** [comment]
Evana, please do a first pass review of this pull request

File: app/components/bootstrap-modal.coffee [hunk 1]
```diff
@@ -12,6 +13,8 @@ BootstrapModalComponent = Ember.Component.extend({
   submitBtnClass: 'btn-primary'
   disableSubmitBtn: false
 
+  submitBtnDisabled: trueNullProperty('disableSubmitBtn')
```
Comment (Evana13G) [inline app/components/bootstrap-modal.coffee]: Why did you add this?

File: app/controllers/index.coffee [hunk 2]
```diff
@@ -15,10 +23,21 @@ IndexController = Ember.Controller.extend({
...
     })
 
+  _updateNotifications: (->
+    @notifications.clearAll()
+    if @get('badURI') || @get('server2BadURI')
+      @notifications.addNotification(
+        message: 'URL not valid. Please enter a valid URL.',
+        type: 'error'
+      )
+  ).observes("badURI", "server2BadURI")
```
Comment (Evana13G) [inline app/controllers/index.coffee]: @schreiaj is it cool to use an observer here? (im scared after your talk...). Should we implement this in another way?

File: app/templates/components/bootstrap-modal.hbs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <div class="modal-dialog modal-center">
   <div class="modal-content">
-    <form {{bind-attr class="formClass"}} {{action 'submit' on='submit'}}>
+    <form class={{formClass}} {{action 'submit' on='submit'}}>
```
Comment (Evana13G) [inline app/templates/components/bootstrap-modal.hbs]: is this change due to the update?

**Comment by Evana13G** [comment]
Everything looks great, see one comment in app/controllers/index.coffee

File: app/controllers/index.coffee (hunk 2 above)
Comment (noranda) [inline app/controllers/index.coffee]: I didn't see your talk, but using observers here is fine.

Comment (schreiaj) [inline app/controllers/index.coffee]: It's perfectly fine but given that you're always clearing the notifications array, is there any reason not to refactor this into using a computed property?

```
badURINotification: Ember.computed('badURI', 'server2BadURI', -> 
  if @get('badURI') || @get('server2BadURI')
    return {message: Blah, type: 'error'}
```

Benefit being that if you don't use it it's not computed. Unless other things are mucking with ```@notifications``` in which case an observer is one approach. The other would be adding the notification wherever the badURI attributes are set.

**Comment by schreiaj** [comment]
Did a quick pass on the code, once we hear back from @noranda about the observer/computed property we can see about merging this.

**Comment by noranda** [comment]
@schreiaj: I've refactored the notifications to not use an observer, take a look when you can.
